utils/digital_interrupts.go

```diff
@@ -57,12 +57,12 @@ func (pull Pull) Validate() error {
 
 // Validate ensures all parts of the config are valid.
 func (config *PinConfig) Validate(path string) error {
-	if config.Name == "" {
-		return resource.NewConfigValidationFieldRequiredError(path, "name")
-	}
 	if config.Pin == "" {
 		return resource.NewConfigValidationFieldRequiredError(path, "pin")
 	}
+	if config.Name == "" {
+		config.Name = config.Pin
+	}
```

Comment:
made name autofill as pin because I kept forgetting it & to match the behavior of the `ByName` apis. you win this time @randhid

Reply:
Should we mutate the config in Validate? I only raise this question because we have less clarity on the config's state as passed into Validate by app, and when being passed into the creator, could we move this logic to the creator?